pkg/controllers/config/cloudinfo.go

```diff
@@ -35,10 +36,20 @@ func isMultiAZDeployment() (bool, error) {
 		}
 	}
 
-	// We consider a cloud multiaz when it either have several different zones
-	// or compute and volumes are different.
-	differentZones := len(ci.ComputeZones) > 0 && len(ci.VolumeZones) > 0 && ci.ComputeZones[0] != ci.VolumeZones[0]
-	return len(ci.ComputeZones) > 1 || len(ci.VolumeZones) > 1 || differentZones, nil
+	// for us to enable the topology feature, we need to have an equal number
+	// of availability zones for the compute and volume services...
+	if len(ci.ComputeZones) != len(ci.VolumeZones) {
+		return false, nil
+	}
+
+	// and these AZs have to have identical names
+	for i := range ci.ComputeZones {
+		if ci.ComputeZones[i] != ci.VolumeZones[i] {
+			return false, nil
+		}
+	}
+
+	return true, nil
```

Comment:
The old code compared just name of the first zone, now all zones are compared. Can it happen that a clusters that had topology enabled will get it disabled in a z-stream update? Is this safe to do?